src/settings.ts

```diff
@@ -6,3 +6,10 @@ export function getConfigFile() {
     undefined
   );
 }
+
+export function getExtendConfiguration(config: string) {
+  return (
+    workspace.getConfiguration('commitlint.config.extend').get(config) ||
+    undefined
+  );
+}
```

Comment:
I think a generic here could improve the typings:
```suggestion
export function getExtendConfiguration<T extends keyof UserConfig>(config: T) {
  return (
    workspace
      .getConfiguration('commitlint.config.extend')
      .get<UserConfig[T]>(config) || undefined
  );
}
```

Then the type assertion wouldn't be necessary where `getExtendConfiguration` gets called.